doc/sphinx/addendum/generalized-rewriting.rst

```diff
@@ -834,17 +868,6 @@ are applied using the tactic :n:`rewrite_strat @rewstrategy`.
    to be documented
 
 
-A few of these are defined in terms of the others using a
-primitive fixpoint operator:
-
-- :n:`try @rewstrategy := choice @rewstrategy id`
-- :n:`any @rewstrategy := fix @ident. try (@rewstrategy ; @ident)`
-- :n:`repeat @rewstrategy := @rewstrategy; any @rewstrategy`
-- :n:`bottomup @rewstrategy := fix @ident. (choice (progress (subterms @ident)) @rewstrategy) ; try @ident`
-- :n:`topdown @rewstrategy := fix @ident. (choice @rewstrategy (progress (subterms @ident))) ; try @ident`
-- :n:`innermost @rewstrategy := fix @ident. (choice (subterm @ident) @rewstrategy)`
-- :n:`outermost @rewstrategy := fix @ident. (choice @rewstrategy (subterm @ident))`
```

Comment:
I moved these without giving them much thought.  In fact, only the one for try looks reasonable.  `fix` is not a rewstrategy.  There is a tactic with that name, but it takes 2 arguments (ident natural) and it begins a proof--that seems incorrect.  `try @ident` doesn't make sense, maybe they should be `try @rewstrategy`.
The ones with semicolons probably should be parenthesized so they aren't parsed as `tactic1 ; tactic2`.
If you give me the right expressions, I will add them.  Otherwise I will leave them as-is.

Reply:
The expressions are fine.  Confusingly enough, due to parsing rules, if you want to sequence a tactic after `rewrite_strat`, I think you need to do `(rewrite_strat foo); bar`.  It's much like `now` in that way.

The `fix` operator is not exposed, so these are pseudo-strategy expressions.  The `try @ident` is invoking `try` on the ident bound by `fix`, which denotes a strategy.

Reply:
Oh, but you dropped the line mentioning the primitive fixpoint operator which isn't actually exposed to the user, which I think makes moving them make things worse.  I think you should either move them back or else leave a note that the definitions using `fix` are using a non-user-accessible primitive just to explain the meaning.